storage/innobase/fil/fil0crypt.cc

```diff
@@ -221,8 +222,8 @@ void fil_space_crypt_cleanup() {
   mutex_free(&crypt_stat_mutex);
 }
 
-fil_space_crypt_t::fil_space_crypt_t(uint new_type, uint new_min_key_version,
-                                     uint new_key_id,
+fil_space_crypt_t::fil_space_crypt_t(uint new_min_key_version, uint new_key_id,
+                                     const char *new_uuid,
                                      fil_encryption_t new_encryption,
```

Comment:
To be able to identify the encryption key id for every table without any additional information. While copying tables this can be very important information (backups). This is the same as with MK encryption. You can identified MK by looking at encryption header. We better follow the same approach here.

Reply:
well, how does it work with MK, fil_space_t doesn't have uuid member?

Reply:
as per slack discussion, the answer is  uuid and master_key_id from page0 are used to decrypt the tablespace key and the unenecrpyted tablespace key is stored in fil_space_t. So there is no nned for UUID member for MK encryption.

In case of KEYRING, theere is no tablespace key and so UUID is necessary to be form the key and decrypt pages. So it has to be stored in every fil_space_t (currently fil_sapce_t->crypt_data)